src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java

```diff
@@ -37,61 +48,23 @@ public List<OrderDetails> getAllOrders() {
 
         List<OrderDetails> list = new ArrayList<>(orders.size());
         for (Order order : orders) {
-            OrderDetails orderDetails = collectOrderDetails(order);
+            OrderDetails orderDetails = orderMapper.toOrderDetails(order);
+            orderDetails.setTotalPrice(getTotalPriceForOrder(orderDetails.getOrderItems()));
```

Comment:
the problem with your NPE exception is in this method. please check what needs to be changed here.

Reply:
changed priority of calling methods